contracts/wallet/BaseWallet.sol

```diff
@@ -117,41 +120,38 @@ contract BaseWallet {
      * @param _value The value of the transaction.
      * @param _data The data of the transaction.
      */
-    function invoke(address _target, uint _value, bytes calldata _data) external moduleOnly returns (bytes memory _result) {
+    function invoke(address _target, uint _value, bytes calldata _data) external virtual moduleOnly returns (bytes memory _result) {
```

Comment:
Why is `invoke()` a virtual method?

Reply:
Because we are overriding it in `FakeWallet` test contract.

Reply:
I see, but I don't really like that we are suggesting that the method can be overriden. I don't find it clean as our intention is for BaseWallet to be a final class.

We could have FakeWallet implement IWallet and duplicate the code of BaseWallet.

Reply:
I tried using the Doppelganger mocking framework but found a problem instantiating it and logged  https://github.com/EthWorks/Doppelganger/issues/7
Ended up with code duplication until a proper solution is found.